src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -971,7 +971,23 @@ private static bool IsIndirectAssignment(BoundAssignmentOperator node)
         {
             var lhs = node.Left;
 
-            Debug.Assert(node.RefKind == RefKind.None || (lhs as BoundLocal)?.LocalSymbol.RefKind == RefKind.Ref,
+            bool IsAssignable(RefKind lhsKind, RefKind rhsKind)
```

Comment:
>IsAssignable [](start = 17, length = 12)

if this is only for asserting, I'd suggest inlining into the Assert call and simply returning false otherwise. Throwing here will not trigger the Debug window while debugging, and will route it to the VS fault handler.